src/screens/ChatInfoScreen/ChatInfoScreen.tsx

```diff
@@ -272,31 +298,56 @@ const ChatInfoScreen = () => {
     return `(${channelInfo?.memberUsers?.length})`;
   };
 
+  const onSaveChangeName = (name) => {
+    setChannelName(name);
+    handleSaveNameChange(name);
+  };
+
   const getHeaderComponent = React.useMemo(() => {
     return (
       <>
-        <AnonymousChatInfoHeader isCenter onPress={goBack} title={channelInfo?.name} />
+        <AnonymousChatInfoHeader isCenter onPress={goBack} title={channelName} />
         <View style={styles.lineTop} />
-        <TouchableOpacity testID="imageClick">
-          <View style={styles.containerPhoto}>{showImageProfile()}</View>
+        <TouchableOpacity
+          testID="imageClick"
+          onPress={handleOnImageClicked}
+          style={styles.containerPhoto}>
+          {isUploadingImage ? (
+            <ActivityIndicator color={COLORS.white} />
+          ) : (
+            showImageProfile(uploadedImage)
+          )}
         </TouchableOpacity>
         <View style={styles.row}>
           <View style={styles.column}>
             <View style={styles.containerGroupName}>
               <Text numberOfLines={1} style={styles.groupName}>
-                {channelInfo?.name}
+                {channelName}
               </Text>
             </View>
             <Text style={styles.dateCreate}>
               Created {moment(channelInfo?.createdAt).format('MM/DD/YYYY')}
             </Text>
           </View>
+          <TouchableOpacity onPress={handleOpenNameChange} style={styles.pencilIconTouchable}>
+            {isUpdatingName ? (
+              <ActivityIndicator />
+            ) : (
+              <IconPencil width={20} height={20} color={COLORS.gray400} />
+            )}
+          </TouchableOpacity>
         </View>
         <View style={styles.lineTop} />
         <Text style={styles.countUser(params?.from)}>Participants {countParticipant()}</Text>
       </>
     );
-  }, [channelInfo]);
+  }, [channelInfo, isUploadingImage, isUpdatingName, channelName]);
+
+  React.useEffect(() => {
+    if (channelInfo) {
+      setChannelName(channelInfo?.name);
+    }
+  }, [JSON.stringify(channelInfo)]);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of `onSaveChangeName` function and the modification of `getHeaderComponent` seem to be well implemented. The new `ModalChangeName` component is also properly used. Just make sure that all these new functions and components are properly used and handled in the rest of the code.

```diff
+  const onSaveChangeName = (name) => {
+    setChannelName(name);
+    handleSaveNameChange(name);
+  };
+
+  const getHeaderComponent = React.useMemo(() => {
+    return (
+      <>
+        <AnonymousChatInfoHeader isCenter onPress={goBack} title={channelName} />
+        ...
+        <TouchableOpacity onPress={handleOpenNameChange} style={styles.pencilIconTouchable}>
+          {isUpdatingName ? (
+            <ActivityIndicator />
+          ) : (
+            <IconPencil width={20} height={20} color={COLORS.gray400} />
+          )}
+        </TouchableOpacity>
+        ...
+      </>
+    );
+  }, [channelInfo, isUploadingImage, isUpdatingName, channelName]);
+
+  React.useEffect(() => {
+    if (channelInfo) {
+      setChannelName(channelInfo?.name);
+    }
+  }, [JSON.stringify(channelInfo)]);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->